File: Count of sum of consecutives - GFG/count-of-sum-of-consecutives.cpp
//{ Driver Code Starts
#include <bits/stdc++.h>
using namespace std;

// } Driver Code Ends
class Solution {
  public:
    int getCount(int n) {
        // code here
        int count = 0;
        for(int i = 1; i*(i+1) < 2*n; i++){
            double a = (1.0 * n - (i*(i+1)/2))/(i+1);
            if(a -(int)a == 0)
                count++;
        }
        return count;
    }
};

//{ Driver Code Starts.
int main() {
    int t;
    cin >> t;
    while (t--) {
        int N;
        
        cin>>N;

        Solution ob;
        cout << ob.getCount(N) << endl;
    }
    return 0;
}
// } Driver Code Ends

File: Count of sum of consecutives - GFG/count-of-sum-of-consecutives.py
#User function Template for python3

class Solution:
    def getCount(self, n):
        # code here 
        count = 0
        i = 1
        while i*(i+1) < n*2:
            a = (n - (i*(i+1)/2))/(i+1)
            if a -int(a) == 0:
                count+=1
            i+=1
        return count

#{ 
 # Driver Code Starts
#Initial Template for Python 3

if __name__ == '__main__': 
    t = int (input ())
    for _ in range (t):
        N=int(input())
        
        ob = Solution()
        print(ob.getCount(N))
# } Driver Code Ends